source/content.js

```diff
@@ -24,6 +24,43 @@ function hideListAddActivity() {
 	$('#stream-items-id .js-activity-list_member_added').css('display', 'none');
 }
 
+function inlineInstagramPosts() {
+	$('a.twitter-timeline-link[data-expanded-url*="//www.instagram.com').each((idx, instagramAnchor) => {
+		const tweetElement = $(instagramAnchor).parents('.js-tweet-text-container');
+		// Don't do anything else if the image is already embedded
+		if ($(tweetElement).siblings('.AdaptiveMedia').length > 0) {
+			return true;
+		}
+		const instagramPostUrl = instagramAnchor.dataset.expandedUrl;
+		const imageRegex = /"display_url": ?"([^"]+)"/g;
+		fetchInstagramPhoto(instagramPostUrl)
+			.then(postHtml => {
+				const matches = imageRegex.exec(postHtml);
+				const instagramImageUrl = matches[1] || null;
+				if (instagramImageUrl) {
+					insertInstagramPhotoInto(tweetElement, instagramImageUrl, instagramPostUrl);
+				}
+			});
+	});
+}
+
+function insertInstagramPhotoInto(element, imageUrl, postUrl) {
```

Comment:
I think this function should only be concerned with creating the element, and have the `element.after` call on line 41.